parquet-column/src/main/java/parquet/schema/Types.java

```diff
@@ -555,6 +622,1024 @@ protected GroupType build(String name) {
+      }
+
+      public ModListBuilder<P> optionalListValue() {
+        return listValue(Type.Repetition.OPTIONAL);
+      }
+    }
+
+    @Override
+    protected abstract Q self();
+
+    @Override
+    protected Type build(String name) {
+      Preconditions.checkNotNull(elementType, "List element type");
+      return buildGroup(repetition)
+          .as(OriginalType.LIST)
+          .repeatedGroup()
+          .addFields(elementType)
+          .named("list")
+          .named(name);
+    }
+
+    public P named(String name) {
+      Preconditions.checkState(originalType == null,
+          "LIST is already the logical type and can't be changed");
+      return super.named(name);
+    }
+
+    public ListGroupElementBuilder<P, Q> groupElement(Type.Repetition repetition) {
+      Preconditions
+          .checkState(elementType == null, "groupElement can be called just once on a ListBuilder");
+      if (parent != null) {
+        return new ListGroupElementBuilder<P, Q>(this).repetition
+            (repetition);
+      } else {
+        return new ListGroupElementBuilder<P, Q>(this, returnType).repetition
+            (repetition);
+      }
+    }
+
+    public ListGroupElementBuilder<P, Q> requiredGroupElement() {
+      return groupElement(Type.Repetition.REQUIRED);
+    }
+
+    public ListGroupElementBuilder<P, Q> optionalGroupElement() {
+      return groupElement(Type.Repetition.OPTIONAL);
+    }
+
+    public ListElementBuilder<P, Q> element(PrimitiveTypeName type, Type.Repetition repetition) {
+      Preconditions
+          .checkState(elementType == null, "groupElement can be called just once on a ListBuilder");
+      if (parent != null) {
+        return new ListElementBuilder<P, Q>(this, type)
+            .repetition(repetition);
+      } else {
+        return new ListElementBuilder<P, Q>(this, returnType, type)
+            .repetition(repetition);
+      }
+    }
+
+    public ListElementBuilder<P, Q> requiredElement(PrimitiveTypeName type) {
+      return element(type, Type.Repetition.REQUIRED);
+    }
+
+    public ListElementBuilder<P, Q> optionalElement(PrimitiveTypeName type) {
+      return element(type, Type.Repetition.OPTIONAL);
+    }
+
+    public ListListElementBuilder<P, Q> listElement(Type.Repetition repetition) {
+      Preconditions
+          .checkState(elementType == null, "groupElement can be called just once on a ListBuilder");
+      if (parent != null) {
+        return new ListListElementBuilder<P, Q>(this).repetition(repetition);
+      } else {
+        return new ListListElementBuilder<P, Q>(this, returnType).repetition(repetition);
+      }
+    }
+
+    public ListListElementBuilder<P, Q> requiredListElement() {
+      return listElement(Type.Repetition.REQUIRED);
+    }
+
+    public ListListElementBuilder<P, Q> optionalListElement() {
+      return listElement(Type.Repetition.OPTIONAL);
+    }
+
+    public ListMapElementBuilder<P, Q> mapElement(Type.Repetition repetition) {
```

Comment:
I think we can allow list of maps. However, the problem is that we do not want to handle lists of maps as keys of a map, this we can catch and advise user to use key(Type) method on MapBuilder instead.

I will make this change and if that does not look good to you then we can get rid of it.